### ISSUE TITLE
Test for WCAG technique G145, G18 and G17: Ensuring that a contrast ratio exists between text (and images of text) and background behind the text

### ISSUE THREAD

**Issue by kevee** [quailjs/quail#71]
The following tests are the same, except the parameters:
http://www.w3.org/TR/WCAG20-TECHS/G145.html
http://www.w3.org/TR/WCAG20-TECHS/G18.html
http://www.w3.org/TR/WCAG20-TECHS/G17.html

G145: contrast ratio 3:1, text size >18pt (AA)
G18: contrast ratio 4,5:1, text size >18pt (AA), or <18pt (AAA)
G17: contrast ratio 7:1 , text size <18pt (AAA)

text size: 18 point (if not bold) or at least 14 point (if bold).
## Description

The objective of this technique is to make sure that users can read text that is presented over a background. 

If the background is a solid color (or all black or all white) then the contrast ratio of the larger-scale text can be maintained by making sure that each of the text letters have a contrast ratio with the background.

If the background or the letters vary in relative luminance (or are patterned), then the background around the letters can be chosen or shaded so that the letters maintain a contrast ratio with the background behind them even if they do not have that contrast ratio with the entire background.

The contrast ratio can sometimes be maintained by changing the relative luminance of the letters as the relative luminance of the background changes across the page.

Another method is to provide a halo around the text that provides the necessary contrast ratio if the background image or color would not normally be sufficiently different in relative luminance.
## QUAIL

This is related to configuration management since we already have color components in quail.
